dashboard/src/repositories/postgres/dashboard.ts

```diff
@@ -42,6 +41,20 @@ export async function findUserDashboard(data: DashboardFindByUserData): Promise<
   }
 }
 
+export async function findUserDashboardWithDashboardOrNull(data: DashboardFindByUserData) {
+  const rel = await prisma.userDashboard.findFirst({
+    where: { dashboardId: data.dashboardId, userId: data.userId },
+    include: { dashboard: true },
+  });
+
+  if (rel === null) return null;
+
+  return {
+    userDashboard: DashboardUserSchema.parse(rel),
+    dashboard: DashboardSchema.parse(rel.dashboard),
```

Comment:
The naming seems a little off to me, a `userDashboard` sounds like a dashboard to me, but there's also a `dashboard`. Should it maybe be:
```suggestion
    dashboardUser: DashboardUserSchema.parse(rel),
    dashboard: DashboardSchema.parse(rel.dashboard),
```

Reply:
Agree. I have made changes to the code, but the schema is named userDashboard, so this entails adding a migration for this rename. Should not be part of this issue.